src/model/Solution.java

```diff
@@ -0,0 +1,34 @@
+package model;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class Solution {
+
+    private ArrayList<State> solution;
+
+    public ArrayList<State> setSolution(State exitPoint) {
+
+        this.solution = new ArrayList<>();
+        while (exitPoint.getParent() != null) {
+            // should handle type of exitPoint:
```

Comment:
What do you mean?

Reply:
!= null due to a warning (have to check its existence), regarding the comment, it isn't clear so I'll change it.